packages/template-retail-react-app/app/hooks/use-update-shopper-context.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
+import {
+    useUsid,
+    useShopperContext,
+    useShopperContextsMutation
+} from '@salesforce/commerce-sdk-react'
+import {isServer, isHydrated} from '@salesforce/retail-react-app/app/utils/utils'
+import {useQueryClient} from '@tanstack/react-query'
+
+// Hooks
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
+
+// Constants
+import {useShopperContextSearchParams} from '@salesforce/retail-react-app/app/hooks/use-shopper-context-search-params'
+
+/*
+ * This hook will set the shopper context when search params pertinant
+ * to shopper context are present.
+ */
+export const useUpdateShopperContext = () => {
+    const {site} = useMultiSite()
+    const {usid} = useUsid()
+    const {search} = useLocation()
+    const queryClient = useQueryClient()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {parameters: {usid, siteId: site.id}},
+        {enabled: !isServer}
+    )
+
+    const refetchDataOnClient = () => {
+        queryClient.invalidateQueries()
+    }
+
+    const handleShopperContextUpdate = async (updateShopperContextObj) => {
+        if (Object.keys(updateShopperContextObj).length === 0) {
+            return
+        }
+
+        const payload = {
+            parameters: {usid, siteId: site.id},
+            body: updateShopperContextObj
+        }
+        if (!shopperContext) {
```

Comment:
`shopperContext` can be initially undefined because the `useShopperContext` call is still loading. So I think the function should exit earlier like this:

```js
    const handleShopperContextUpdate = async (updateShopperContextObj) => {
        if (isLoading || Object.keys(updateShopperContextObj).length === 0) {
            return
        }
        ...
    }
```
... where `isLoading` comes from `const {data: shopperContext, isLoading} = useShopperContext(...)`